lib.rs

```diff
@@ -87,16 +140,24 @@ impl<D: Decorator> Format<D> {
         let r_decorator = self.decorator.decorate(record);
 
 
-        try!(self.print_msg_header(io, &r_decorator, record));
+        let mut skip_msg_comma = try!(self.print_msg_header(io, &r_decorator, record));
```

Comment:
Even if the variable were a member variable, I think there'd still be the issue of `Fn` versus `FnOnce`. I'll test it out to make sure though.